### PR TITLE
Frontend: com_search view review

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Frontend: com_search view review

### Testing Instructions

Install testing data and test the frontend com_search

### Expected result

works

### Actual result

works, with cs issues

### Documentation Changes Required

none

### CODE REVIEW

File: components/com_search/views/search/tmpl/default_form.php [hunk 1]
```diff
@@ -48,33 +46,27 @@
...
-				$checked = is_array($this->searchareas['active']) && in_array($val, $this->searchareas['active']) ? 'checked="checked"' : '';
-			?>
-			<label for="area-<?php echo $val; ?>" class="checkbox">
-				<input type="checkbox" name="areas[]" value="<?php echo $val; ?>" id="area-<?php echo $val; ?>" <?php echo $checked; ?> >
-				<?php echo JText::_($txt); ?>
-			</label>
+			<?php foreach ($this->searchareas['search'] as $val => $txt) : ?>
+				<?php $checked = is_array($this->searchareas['active']) && in_array($val, $this->searchareas['active']) ? 'checked="checked"' : ''; ?>
+				<label for="area-<?php echo $val; ?>" class="checkbox">
+					<input type="checkbox" name="areas[]" value="<?php echo $val; ?>" id="area-<?php echo $val; ?>" <?php echo $checked; ?> >
```
Comment (n9iels) [inline components/com_search/views/search/tmpl/default_form.php]: This line should be closed with `/>` (If I see it right)

File: components/com_search/views/search/tmpl/default_results.php [hunk 2]
```diff
@@ -11,35 +11,34 @@
...
-<?php foreach ($this->results as $result) : ?>
-	<dt class="result-title">
-		<?php echo $this->pagination->limitstart + $result->count . '. '; ?>
-		<?php if ($result->href) : ?>
-			<a href="<?php echo JRoute::_($result->href); ?>"<?php if ($result->browsernav == 1) : ?> target="_blank"<?php endif; ?>>
+	<?php foreach ($this->results as $result) : ?>
+		<dt class="result-title">
+			<?php echo $this->pagination->limitstart + $result->count . '. '; ?>
+			<?php if ($result->href) : ?>
+				<a href="<?php echo JRoute::_($result->href); ?>"<?php if ($result->browsernav == 1) : ?> target="_blank"<?php endif; ?>>
```
Comment (n9iels) [inline components/com_search/views/search/tmpl/default_results.php]: Maybe a inline if statement here? Don't know if that is valid by Joomla! Codestyle.

File: components/com_search/views/search/tmpl/default_error.php [hunk 3]
```diff
@@ -11,7 +11,7 @@
 ?>
 
 <?php if ($this->error) : ?>
-<div class="error">
+	<div class="error">
 			<?php echo $this->escape($this->error); ?>
```
Comment (Quy) [inline components/com_search/views/search/tmpl/default_error.php]: Remove tab.

File: components/com_search/views/search/tmpl/default_form.php [hunk 4]
```diff
@@ -26,13 +26,11 @@
...
 	<div class="searchintro<?php echo $this->params->get('pageclass_sfx'); ?>">
 		<?php if (!empty($this->searchword)) : ?>
-		<p><?php echo JText::plural('COM_SEARCH_SEARCH_KEYWORD_N_RESULTS', '<span class="badge badge-info">' . $this->total . '</span>'); ?></p>
+			<p><?php echo JText::plural('COM_SEARCH_SEARCH_KEYWORD_N_RESULTS', '<span class="badge badge-info">' . $this->total . '</span>'); ?></p>
 		<?php endif; ?>
 	</div>
-
 	<?php if ($this->params->get('search_phrases', 1)) : ?>
 		<fieldset class="phrases">
 			<legend><?php echo JText::_('COM_SEARCH_FOR'); ?>
```
Comment (Quy) [inline components/com_search/views/search/tmpl/default_form.php]: Split into multiple lines.

**Comment by zero-24** [comment]
ok so finally i have fixed the merge conflicts here too.

File: components/com_search/views/search/tmpl/default_error.php [hunk 5]
```diff
@@ -8,10 +8,10 @@
  */
 
 defined('_JEXEC') or die;
-?>
 
+?>
 <?php if ($this->error) : ?>
-<div class="error">
+	<div class="error">
 			<?php echo $this->escape($this->error); ?>
```
Comment (Quy) [inline components/com_search/views/search/tmpl/default_error.php]: Remove tab

File: components/com_search/views/search/tmpl/default_form.php [hunk 6]
```diff
@@ -26,13 +26,11 @@
 		<input type="hidden" name="task" value="search" />
```
Comment (Quy) [inline components/com_search/views/search/tmpl/default_form.php:29]: Line 24: Split into multiple lines.

File: components/com_search/views/search/tmpl/default_form.php [hunk 7]
```diff
@@ -48,33 +46,27 @@
 				</div>
```
Comment (Quy) [inline components/com_search/views/search/tmpl/default_form.php]: Line 39: Indent

Lines 42-45: Indent

File: components/com_search/views/search/tmpl/default_form.php [hunk 8]
```diff
@@ -48,33 +46,27 @@
...
 		</fieldset>
 	<?php endif; ?>
-
 	<?php if ($this->params->get('search_areas', 1)) : ?>
 		<fieldset class="only">
 			<legend><?php echo JText::_('COM_SEARCH_SEARCH_ONLY'); ?></legend>
```
Comment (Quy) [inline components/com_search/views/search/tmpl/default_form.php]: Split into multiple lines?

File: components/com_search/views/search/tmpl/default_form.php [hunk 9]
```diff
@@ -26,13 +26,11 @@
 		<input type="hidden" name="task" value="search" />
 		<div class="clearfix"></div>
 	</div>
-
 	<div class="searchintro<?php echo $this->params->get('pageclass_sfx'); ?>">
 		<?php if (!empty($this->searchword)) : ?>
-		<p><?php echo JText::plural('COM_SEARCH_SEARCH_KEYWORD_N_RESULTS', '<span class="badge badge-info">' . $this->total . '</span>'); ?></p>
+			<p><?php echo JText::plural('COM_SEARCH_SEARCH_KEYWORD_N_RESULTS', '<span class="badge badge-info">' . $this->total . '</span>'); ?></p>
```
Comment (Quy) [inline components/com_search/views/search/tmpl/default_form.php]: Split into multiple lines?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 0cc0a8c5ead18a04c81df4f4178b20b0fecd5f8d<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17234">issues.joomla.org/tracker/joomla-cms/17234</a>.</sub>

**Comment by sanderpotjer** [comment]
I have tested this item :red_circle: unsuccessfully on 0cc0a8c5ead18a04c81df4f4178b20b0fecd5f8d<br /><br />After applying the patch the category is no longer visible in the search results:

Before:
screen shot 2017-09-02 at 11 15 36

After:
screen shot 2017-09-02 at 11 15 28

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17234">issues.joomla.org/tracker/joomla-cms/17234</a>.</sub>

**Comment by zero-24** [comment]
fixed with the last commit @sanderpotjer thanks 👍

**Comment by sanderpotjer** [comment]
I have tested this item :white_check_mark: successfully on bf83d4922a82b8ad1b2df1f88e2201cc9c39d002<br /><br />@zero-24 thanks for the fix, all good now!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17234">issues.joomla.org/tracker/joomla-cms/17234</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please retest?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bf83d4922a82b8ad1b2df1f88e2201cc9c39d002<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17234">issues.joomla.org/tracker/joomla-cms/17234</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Pull request merged**